torch/quantization/fx/_equalize.py

```diff
@@ -298,6 +322,78 @@ def maybe_get_weight_eq_obs_node(op_node: Node, modules: Dict[str, nn.Module]) -
             return node_arg
     return None
 
+def maybe_get_following_observer(
+    node: Node,
+    modules: Dict[str, nn.Module],
+    target_observer_type: Any = object,
+) -> Optional[Node]:
+    """ Loops through the graph forwards and finds the next observer node with
+    the given type. We will skip nodes such as relu and pooling.
+
+    For example if we had the following part of the graph:
+        linear1 -> relu -> inp_quant_obs -> pool -> inp_eq_obs -> linear2
+    If we are looking for the quantization observer, we want to start from the
+    given node (linear1) and traverse forwards skipping the relu layer, and
+    returning the quantization observer.
+    If we are looking for the equalization observer, we want to start from the
+    given node (inp_quant_obs) and traverse forwards skipping the pooling layer,
+    and returning the equalization observer.
+    """
+
+    temp_node = node
+    while len(temp_node.users.items()) > 0:
+        possible_node = None
+        for user, _ in temp_node.users.items():
+            if user.op == 'call_module' and isinstance(modules[str(user.target)], target_observer_type):
+                # If the node has the observer type specified, return the obsever node
+                return user
+            if user.op in {'call_module', 'call_function', 'call_method'}:
+                possible_node = user
+                break
+
+        if possible_node is None or not is_skippable_node(possible_node, modules):
+            return None
+        temp_node = possible_node
+
+    return None
+
+def maybe_get_prev_node(node, modules):
+    """ Loops through the graph backwards and finds the previous op node. We
+    will skip nodes such as relu and pooling.
+
+    For example if we had the following part of the graph:
+        linear1 -> relu -> inp_quant_obs -> pool -> inp_eq_obs (node) -> linear2
+    We want to start from the given node (inp_eq_obs) and traverse backwards
+    skipping the pooling layer, finding the quantization observer, skipping the
+    relu layer, and finally finding the linear1 op node.
+
+    Returns:
+        - The previous op node if it exists
+        - The previous quantization observer node that appears before the
+        previous op node if it exists
+    """
+
+    inp_quant_obs_node = None
+    prev_node = None
+
+    temp_node = node
+    while len(temp_node.args) > 0:
+        possible_node = temp_node.args[0]
```

Comment:
I feel this loop is a little bit confusing, can we just assign `temp_node = temp_node.args[0]`here and redo the condition checks below?

looks like in all three branches we can do `temp_node = temp_node.args[0]`